### ISSUE TITLE
No lightweight way to adjust ui style

### ISSUE THREAD

**Issue by vkkoskie** [emilk/egui#17]
I could be wrong about the following. I'm just starting to learn Rust, so I could have misunderstood something. I believe this is the only way to change style elements given the current API:

    let mut style = ui.style().clone();
    style.item_spacing = egui::vec2(x, y);
    ui.set_style(style);

The `ui`'s `style` member is private and the `ui.style()` method returns an immutable reference. So the only way to change a single style element is to clone the whole struct out, make the change, and pass ownership back into the `ui`. This is awkward to implement, and forces needless copies.

The patterns in the `Style::ui` widget hints that some kind of public access is safe and may have been intended, but I can't find a pathway for it.

**Comment by emilk** [comment]
Good point! I think allowing `ui.style_mut().item_spacing = vec2(x, y);` would make sense.

**Comment by vkkoskie** [comment]
Great! PR submitted. After submitting, I noticed the same thing about ui.layout. The PR includes both changes. Sorry if that's bad form.
